homeassistant/components/ads/__init__.py

```diff
@@ -140,45 +144,44 @@ def handle_write_data_by_name(call: ServiceCall) -> None:
     "NotificationItem", "hnotify huser name plc_datatype callback"
 )
 
+# Tuple to hold data needed to restore notification
+DeletedNotification = namedtuple("DeletedNotification", "name plc_datatype callback")
+
+
+class ConnectionState(Enum):
+    """Reresentation of ADS connection state."""
+
+    Connected = 1
+    ReadyToReconnect = 2
+    Disconnected = 3
+
 
 class AdsHub:
     """Representation of an ADS connection."""
 
-    def __init__(self, ads_client):
+    def __init__(self, ads_client: pyads.Connection) -> None:
```

Comment:
If you're going to add typing, please also do this for the underlying functions.